instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -44,6 +54,33 @@ def query(sql, options = {})
 
           private
 
+          def obfuscate_sql(sql)
+            return sql unless config[:enable_sql_obfuscation]
+
+            if sql.size > 2000
+              'SQL query too large to remove sensitive data ...'
+            else
+              obfuscated = sql.gsub(generate_mysql_regex, '?')
+              obfuscated = 'Failed to obfuscate SQL query - quote characters remained after obfuscation' if detect_unmatched_pairs(obfuscated)
+              obfuscated
+            end
+          end
+
+          def generate_mysql_regex
+            components = %i[single_quotes double_quotes numeric_literals boolean_literals
+                            hexadecimal_literals comments multi_line_comments]
+            Regexp.union(components.map { |component| COMPONENTS_REGEX_MAP[component] })
+          end
```

Comment:
This should be cached so we don't have to recompile the regex on every call.

Reply:
done 👍